back-nest/src/getAllElement/products.contoller.ts

```diff
@@ -0,0 +1,13 @@
+import { Controller, Get } from '@nestjs/common';
+import { ProductService } from './products.services';
+import { Product } from '../models/product.enity.model';
+
+@Controller('product')
+export class ProductController {
+  constructor(private readonly productService: ProductService) {}
+
+  @Get()
+  async getAllProducts(): Promise<Product[]> {
+    return await this.productService.getAllProducts();
```

Comment:
You don't need to return an awaited promise. In such cases `return promise` is enough

```suggestion
    return this.productService.getAllProducts();
```